src/classes/Gift.cls

```diff
@@ -0,0 +1,89 @@
+/*
+    Copyright (c) 2021 Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2021
+* @description Gift entity class in NPSP.
+*/
+
+public with sharing class Gift {
+    
+    @TestVisible
+    private DataImport__c dataImport;
+    private List<String> availableFields;
+    private GiftSelector giftSelector {
+        get {
+            if (giftSelector == null) {
+                giftSelector = new GiftSelector();
+            }
+            return giftSelector;
+        }
+        set;
+    }
+    
+    public Gift(DataImport__c dataImport) {
+        this.dataImport = dataImport;
+        this.populateAvailableFields();
+    }
+
+    public Boolean isGiftReauthorizedRequired() {
```

Comment:
This should probably just be named "isAuthorized()".  Reauth required is dependent on context.

Reply:
Feel free to push back against this... just wondering if `isAuthorized()` makes the most sense for this method in its current structure. 

To me, a gift being authorized is a state and preventing an edit of an authorized gift is a constraint. One is certainly dependent on the other. My concern is that if we move to `isAuthorized`, we lose resolution to one of those items. A gift can be authorized. At the same time, we may or may not want to prevent that authorized gift from being edited depending on the scenario.

I feel like what you're proposing tracks against the notion of a given "type" being more than just a data structure, but also having behavior and constraints baked into it. Perhaps this is a larger conversation on the Gift entity as a whole? Happy to chat about this more.

Reply:
What about something like "allowsChange()" or something better but still includes the specific context in the name? Or maybe even "requiresReauthorizationBeforeChange()" or something?

Reply:
`requiresReauthorizationBeforeChange()` feels like the winner to me. Let me go ahead and swap out the methods name right now and then we can always adjust to something else anytime thereafter if need be.